### PR TITLE
Added Painter

### PR DESCRIPTION
**Pull request by IDF31**
Painter is a widget based on Tkinter Canvas, made for easy drawing.

### CODE REVIEW

**Comment by IDF31** [comment]
I made a Pong clone using my modifications, just for an example

`from guizero import App, Painter

app = App()
p = Painter(app,width=500,height=500,background='black')

y = 250
x = 10

bx,by = 250,250
vx,vy = 10,10

def main():
    global x,y,bx,by,vx,vy

    bx+=vx
    by+=vy

    if bx+10 >= 500: vx=-vx
    if by+10 >= 500 or by <= 0: vy=-vy
    if x <= bx <= x+10 and y <= by <= y+250/2: vx=-vx
    if bx <= 0: bx,by = 250,250
    p.clear()
    p.draw_rectangle(10,y,10,150,color='white')
    p.draw_ellipse(bx,by,10,10,color='white')

def move(event):
    global x,y
    
    if event.keysym == 'Up': y-=15
    if event.keysym == 'Down': y+=15
    
app.on_key_press(move)
app.repeat(1,main)
`

File: guizero/Painter.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from tkinter import Canvas, ALL
+from . import utilities as utils
+
+
+class Painter:
```
Comment (martinohanlon) [inline guizero/Painter.py]: The painter class should inherit the mixins it implements (Container, Widget, etc) and the tkmixins Canvas suports (Enable, Display, Schedule, etc).

File: guizero/Painter.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    def __init__(self,master,width=100,height=100,background='white',grid=None,align=None):
+
+        self.tk = Canvas(master.tk,width=width,height=height,background=background)
+        self._height = height
+        self._width = width
+
```
Comment (martinohanlon) [inline guizero/Painter.py]: the class should preserve its master, grid and align attributes, so they are accessible to guizero through the mixins

File: guizero/Painter.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        self.tk = Canvas(master.tk,width=width,height=height,background=background)
+        self._height = height
+        self._width = width
+        
+        utils.auto_pack(self,master,grid,align)
+
+    def draw_line(self,x1,y1,x2,y2,color='black'):
+        self.tk.create_line(x1,y1,x2,y2,fill=color)
+
+    def draw_ellipse(self,x,y,width,height,color=''):
```
Comment (martinohanlon) [inline guizero/Painter.py]: should color have a default?

File: guizero/Painter.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+from tkinter import Canvas, ALL
+from . import utilities as utils
+
+
+class Painter:
+    def __init__(self,master,width=100,height=100,background='white',grid=None,align=None):
```
Comment (martinohanlon) [inline guizero/Painter.py]: `background` should be persisted as an attribute, named `bg` and exposed as a property to be consistent with other widgets.

File: guizero/Painter.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    def draw_ellipse(self,x,y,width,height,color=''):
+        self.tk.create_oval(x,y,x+width,y+height,fill=color)
+
+    def draw_rectangle(self,x,y,width,height,color=''):
+        self.tk.create_rectangle(x,y,x+width,y+height,fill=color)
+
+    def clear(self):
+        self.tk.delete(ALL)
+
+
```
Comment (martinohanlon) [inline guizero/Painter.py]: Painter should implement `enabled`, `enable` and `disable` - have a look at `Waffle` its a similar widget.

**Comment by IDF31** [comment]
Closing this PR because I reforked the repo

**Pull request closed without merging**